testutil/checkers_test.go

```diff
@@ -20,52 +20,86 @@
 package testutil
 
 import (
-	. "gopkg.in/check.v1"
+	"reflect"
 	"testing"
+
+	. "gopkg.in/check.v1"
 )
 
 func Test2(t *testing.T) {
 	TestingT(t)
 }
 
-type CheckersSuite struct{}
+type CheckersS struct{}
+
+var _ = Suite(&CheckersS{})
 
-var _ = Suite(&CheckersSuite{})
+func testInfo(c *C, checker Checker, name string, paramNames []string) {
+	info := checker.Info()
+	if info.Name != name {
+		c.Fatalf("Got name %s, expected %s", info.Name, name)
+	}
+	if !reflect.DeepEqual(info.Params, paramNames) {
+		c.Fatalf("Got param names %#v, expected %#v", info.Params, paramNames)
+	}
+}
+
+func testCheck(c *C, checker Checker, result bool, error string, params ...interface{}) ([]interface{}, []string) {
+	info := checker.Info()
+	if len(params) != len(info.Params) {
+		c.Fatalf("unexpected param count in test; expected %d got %d", len(info.Params), len(params))
+	}
+	names := append([]string{}, info.Params...)
+	resultActual, errorActual := checker.Check(params, names)
+	if resultActual != result || errorActual != error {
+		c.Fatalf("%s.Check(%#v) returned (%#v, %#v) rather than (%#v, %#v)",
+			info.Name, params, resultActual, errorActual, result, error)
+	}
+	return params, names
+}
 
-func (s *CheckersSuite) TestUnsupportedTypes(c *C) {
-	c.ExpectFailure("haystack is of unsupported type int")
-	c.Assert(5, Contains, "foo")
+func (s *CheckersS) TestUnsupportedTypes(c *C) {
```

Comment:
The helper is cherry-picked from upstream. I've used it so that it is easier to integrate. Originally I wrote tests just like yours but I've re-written them after Gustavo's advice.